doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_clockman.rst

```diff
@@ -0,0 +1,111 @@
+* HSFLL Secure
+* HSFLL Global
+
+Each clock domain is enabled independently of others and it is automatically enabled when at least one sink is active.
+The firmware can choose to keep each clock domains enabled even when all its sinks are inactive.
+
+Clock domain source selection differs between clock domains:
+
+* The LFCLK source is selected by the System Controller Firmware.
+* The HFXO source is selected by the System Controller Firmware.
+* The FLL16M source is selected by local domains or System Controller Firmware (SCFW) in all scenarios, except one:
+  When the firmware selects open-loop, but any of the HSFLL sources is in closed-loop, the FLL16M is automatically switched by the hardware to closed-loop.
+
+* HSFLLs sources are selected by firmware (depending on an HSFLL instance: local domain or System Controller).
+
+The application-facing APIs expose only the domain that directly clocks the component used by the application.
+The management of the clocks on which this domain depends is handled internally by the clock driver associated with this domain.
+
+For example, a firmware module might request better timing accuracy for a fast UART instance.
+In this case, the firmware module requests the accuracy from the global HSFLL device driver, which directly clocks the UART.
+The dependencies, such as FLL16M and LFXO, are managed internally by the global HSFLL driver, not directly by the firmware module.
+
+
+LFCLK
+=====
+
+The Low-Frequency Clock domain is responsible for generating a 32768 Hz clock signal for ultra-low-power peripherals like ``GRTC`` or ``RTC``.
+
+HFXO
+====
+
+The High-Frequency Crystal Oscillator domain is responsible for clocking peripherals requiring high accuracy and MHz range frequency.
+The drawback of this clock source is relatively high power consumption so it is enabled only when needed.
+
+Hardware capabilities
+---------------------
+
+The HFXO, like any other clock domain, can be connected in hardware to one of the available sources.
+And it provides the clock signal to all of its sinks.
+
+FLL16M
+======
+
+The FLL16M clock domain clocks most of the peripherals in the system ("slow", 16 MHz).
+Its accuracy results in the timing accuracy of slow ``UART``, ``SPI``, ``TWI``, ``PWM``, ``SAADC``, etc.
+
+Local HSFLLs
+============
+
+Local HSFLLs are clocking CPUs in local domains, fast peripherals around them, and local RAM.
+
+Global HSFLL
+============
+
+Global HSFLL clocks "fast" peripherals, global CPUs, RAM, and MRAM blocks.
+
+Zephyr clock control API
+************************
+
+Zephyr RTOS contains a ``clock_control`` device driver class for managing clocks.
+The ``clock_control`` API is designed to support multiple requestors.
+Moreover, the ``clock_control`` API supports blocking or asynchronous operations based on notifications when the requested clock settings are applied.
+Because of this, the ``clock_control`` API is well-suited for LFCLK management in the local domains.
```

Comment:
```suggestion
```

This sentence is a justification of the design decision. It doesn't provide user-related information.